creusot/src/backend/dependency.rs

```diff
@@ -40,6 +40,7 @@ impl<'tcx> Dependency<'tcx> {
                 TyKind::Adt(def, substs) => Some((def.did(), substs)),
                 TyKind::Closure(id, substs) => Some((*id, substs)),
                 TyKind::Alias(AliasTyKind::Projection, aty) => Some((aty.def_id, aty.args)),
+                TyKind::Alias(AliasTyKind::Opaque, aty) => Some((aty.def_id, aty.args)),
```

Comment:
I guess you could factorize too?

Reply:
I guess we could also replace the assertion with a pattern checking that the alias kind is either opaque or a projection?